app/views/shared/_draft_mode_notice.html.erb

```diff
@@ -0,0 +1,26 @@
+<% if app.draft? %>
```

Comment:
I usually do this high-level check outside the partial. so that the places where partials are called from are more readable.